CivilizationMapParser/Program.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace CivilizationMapParser
+{
+	class Program
+	{
+		static void Main(string[] args)
+		{
+			ReadSourceFile(AskForPath());
+			WriteRegionTileInfo();
+		}
+
+		public static List<RegionTile> Regions = new List<RegionTile>();
+
+		public static string AskForPath()
+		{
+			string path;
+			Console.WriteLine("Provide path to Source file below:");
+			return path = Console.ReadLine();
+		}
+
+		public static void ReadSourceFile(string path)
+		{
+			string[] lines = File.ReadAllLines(path);
+			RegionTile Region = null;
+
+			for (int i = 0; i < lines.Length; i++)
+			{
+				if (Region == null)
+				{
+					Region = new RegionTile();
+				}
+				else if (!String.IsNullOrEmpty(Region.Name) && 
+						Region.NumberOfVectors != 0 && 
+						Region.Vertices != null &&
+						Region.Triangles != null &&
+						!Regions.Contains(Region))
+				{
+					Regions.Add(Region);
+					Region = new RegionTile();
+				}
+
+				if (Region != null && lines[i].Contains("tile.name = "))
+				{
+					string name = lines[i].Substring(lines[i].IndexOf('"'));
+					name = name.Trim(' ', ';', '"');
+					Region.Name = name;
+
+					Console.WriteLine();
+					Console.WriteLine(Region.Name);
+					Console.WriteLine();
+				}
+				if (Region != null && lines[i].Contains("int[] tris = new int[]"))
+				{
+					string trianglesString = lines[i].Substring(32);
+					trianglesString = trianglesString.Trim(' ', '}', '{', ';').Replace(" ", "");
+					Console.WriteLine(trianglesString);
+					string[] numAsStr = trianglesString.Split(',');
+
+					Region.Triangles = new List<int>();
```

Comment:
why does class Program initialize properties of the Region object? It should be done in the constructor of Region class. Otherwise it violates Single Responsibility principle